app/controllers/application_controller.rb

```diff
@@ -1,2 +1,15 @@
 class ApplicationController < ActionController::Base
+  include SessionsHelper
+  include CategoriesHelper
+
+  before_action :set_locale
+  protect_from_forgery with: :exception
+
+  def default_url_options
+    {locale: I18n.locale}
+  end
+  private
+  def set_locale
+    I18n.locale = params[:locale] || I18n.default_locale
+  end
```

Comment:
Thêm các line trống vào e nha
```suggestion
  def default_url_options
    {locale: I18n.locale}
  end
  
  private
  
  def set_locale
    I18n.locale = params[:locale] || I18n.default_locale
  end
```